src/services/createRouterReject.ts

```diff
@@ -3,23 +3,22 @@ import { genericRejection } from '@/components/rejection'
 import { createResolvedRouteQuery } from '@/services/createResolvedRouteQuery'
 import { RegisteredRejectionType } from '@/types/register'
 import { ResolvedRoute } from '@/types/resolved'
+import { RouterOptions } from '@/types/router'
 
 export const builtInRejections: ['NotFound'] = ['NotFound']
 export type BuiltInRejectionType = typeof builtInRejections[number]
 
-export type RouterRejectionType = BuiltInRejectionType | RegisteredRejectionType
+export type RouterSetReject = (type: RegisteredRejectionType | null) => void
 
-export type RouterRejectionComponents = { rejections?: Partial<Record<RouterRejectionType, Component>> }
+export type RejectionType<TOptions extends RouterOptions> = keyof TOptions['rejections'] | BuiltInRejectionType
 
-export type RouterSetReject = (type: RouterRejectionType | null) => void
-
-type GetRejectionRoute = (type: RouterRejectionType) => ResolvedRoute
+type GetRejectionRoute = (type: RegisteredRejectionType) => ResolvedRoute
 type IsRejectionRoute = (route: ResolvedRoute) => boolean
 
-export type RouterRejection = Ref<null | { type: RouterRejectionType, component: Component }>
+export type RouterRejection = Ref<null | { type: RegisteredRejectionType, component: Component }>
 
 type CreateRouterRejectContext = {
-  rejections?: RouterRejectionComponents['rejections'],
+  rejections?: Partial<Record<string, Component>>,
```

Comment:
Is this partial needed?

Reply:
yes and no. It adds `undefined` as a possible value which is what we want. I could make it `Record<string, Component | undefined>` if you like that more

Reply:
Makes sense, fine as is